src/directives/float-label/float-label.directive.ts

```diff
@@ -1,18 +1,49 @@
 import { Directive, ElementRef, HostBinding, Input, OnChanges, OnDestroy, OnInit, Renderer2 } from '@angular/core';
+import { AutofillMonitor } from '@angular/cdk/text-field';
+import { Subscription } from 'rxjs';
 
 @Directive({
     selector: '[uxFloatLabel]',
     host: {
         'class': 'ux-float-label'
     }
 })
-export class FloatLabelDirective implements OnInit, OnChanges, OnDestroy {
+export class FloatLabelDirective<T = string> implements OnInit, OnChanges, OnDestroy {
 
     @Input('uxFloatLabel')
-    input: HTMLInputElement;
+    set input(input: HTMLInputElement) {
+
+        // remove any previous autofill subscriptions
+        if (this._input) {
+            this._autofillMonitor.stopMonitoring(this._input);
+        }
+
+        this._subscription.unsubscribe();
+
+        this._input = input;
+
+        // if the input is null then don't need to subscribe to autofillMonitor
+        if (!input) {
+            return;
+        }
+
+        // create a new autofillMonitor subscription
+        this._subscription = this._autofillMonitor.monitor(input).subscribe(event => {
+            if (!this.raised && event.isAutofilled) {
+                this.raised = true;
+            }
+            if (this.raised && !event.isAutofilled) {
```

Comment:
I wonder could you check this for me? This condition says that it the label is raised and we remove autofill we should set raised to false. 

What would happen if the user typed in `t` (this would raise the text), then auto fill offers a suggestion for `tara.todd@microfocus.com` which you hover over (isAutofilled now true), now if I hover out of the autofill (isAutofill now false), but there is text in the input so raised should remain true. I suspect we need to check if there is no text in the input before setting raised to false